### DISCUSSION TITLE
Backtesting strategies acting on order book

### DISCUSSION THREAD

**Discussion by Kayne88** [kernc/backtesting.py#566] [Q&A] [answered]
I am sure this question has already been stated, but I was unable to locate it.

I was wondering, if backtesting lib is supporting the backtest of strategies using the order book (for example 250ms ticks)? If this is not the case, what would be the steps to extend the current lib to account for this need?

Thinking about this, I would choose a naive approach of loading order book tick data instead of ohlc data and reveal them in next(). What would be the implication of such an approach? Could the lib handle such an approach "out-of-the-box"?

I am also a little worried about performance, when feeding tick data. Backtests already take a long time on 1m ohlc data.

Happy for inputs.

**Comment by kernc** [comment, marked as answer]
You could pass extra columns (such as columns of bid/ask order book objects) in the [input `data` DataFrame](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest) to base orders upon, but if your strategy is such that it would in fact considerably influence the state of the order book (such as arbitrage), then I don't think this is the right framework. Nor would I recommend backtesting on tick data, but to have it, set `O=H=L=C=price`.
